dev/com.ibm.ws.org.apache.cxf.cxf.core.3.2/src/org/apache/cxf/message/MessageImpl.java

```diff
@@ -1154,12 +1045,21 @@ public Object replace(String key, Object value) {
     public boolean replace(String key, Object oldValue, Object newValue) {
         Integer index = KEYMAP.getOrDefault(key, KEY_NOT_FOUND);
         if (index != KEY_NOT_FOUND) {
-            if (propertyValues[index] == oldValue) {
-                propertyValues[index] = newValue;
-                return true;
-            } else {
+            if (propertyValues[index] != NOT_FOUND) {
+                if (propertyValues[index] == null) {
+                    if (oldValue == null) {
+                        propertyValues[index] = newValue;
+                        return true;
+                    } else {
+                        return false;
+                    }
+                } else if (propertyValues[index].equals(oldValue)){
+                    propertyValues[index] = newValue;
+                    return true;
+                }
```

Comment:
Again I recommend doing the == null check on oldValue instead of proeprtyValues[index] and then do oldValue.equals to reverse the equals.

Reply:
Same with remove with check Value.